File: app_macOS-Intel_BridgeRestore/scripts/bridge-restore-panel.swift
#!/usr/bin/env swift
// bridge-restore-panel.swift — Phase 1b persistent NSPanel
// Run: swift /path/to/bridge-restore-panel.swift
// VERSION: v1.0.6 (Build 26A06)

import Cocoa
import Foundation

let SCRIPTS = ProcessInfo.processInfo.environment["BR_SCRIPTS"]
    ?? "/Users/akmacks/dev/projects/apps/app_macOS-Intel_BridgeRestore/scripts"
let STATUS_CMD  = "bash \(SCRIPTS)/status-check.sh"
let REFRESH_S: TimeInterval = 15
let W: CGFloat = 840; let H: CGFloat = 520
let LW: CGFloat = 360; let PAD: CGFloat = 12

// ── Palette ───────────────────────────────────────────────────────
func rgb(_ r: CGFloat,_ g: CGFloat,_ b: CGFloat) -> NSColor {
    NSColor(red: r/255, green: g/255, blue: b/255, alpha: 1)
}
let cBg     = rgb(28,28,30);    let cPanel  = rgb(44,44,46)
let cGreen  = rgb(48,209,88);   let cYellow = rgb(255,214,10)
let cRed    = rgb(255,69,58);   let cGrey   = rgb(142,142,147)
let cBlue   = rgb(10,132,255);  let cWhite  = rgb(255,255,255)
let cDim    = rgb(180,180,185); let cBorder = rgb(72,72,74)
let cOutBg  = rgb(20,20,22)

// ── Shell helper ──────────────────────────────────────────────────
func shell(_ cmd: String) -> String {
    let t = Process(); let p = Pipe()
    t.launchPath = "/bin/zsh"; t.arguments = ["-c", cmd]
    t.standardOutput = p; t.standardError = Pipe()
    t.launch(); t.waitUntilExit()
    return String(data: p.fileHandleForReading.readDataToEndOfFile(),
                  encoding: .utf8) ?? ""
}

// ── Parse status JSON ─────────────────────────────────────────────
struct Check: Decodable { let status, value, detail: String }
struct Summary: Decodable { let all_green: Int; let red_count: Int; let grey_count: Int }
struct Status: Decodable {
    let role, friendly, model: String
    let checks: [String: Check]
    let summary: Summary
}
func getStatus() -> Status? {
    guard let d = shell(STATUS_CMD).data(using: .utf8) else { return nil }
    return try? JSONDecoder().decode(Status.self, from: d)
}

// ── UI helpers ────────────────────────────────────────────────────
func makeLabel(_ s: String, _ f: NSFont, _ c: NSColor) -> NSTextField {
    let t = NSTextField(labelWithString: s)
    t.font = f; t.textColor = c
    t.backgroundColor = .clear; t.isSelectable = false
    return t
}
func dotChar(_ s: String) -> String {
    switch s { case "green": return "●"; case "yellow": return "◐"
               case "red":   return "●"; default: return "○" }
}
func dotColor(_ s: String) -> NSColor {
    switch s { case "green": return cGreen; case "yellow": return cYellow
               case "red":   return cRed;   default: return cGrey }
}

// ── Row order ─────────────────────────────────────────────────────
let ROW_KEYS = ["host_ip","bridge","peer_ip","internet","tunnel","controlmaster","is"]
let ROW_LABELS = ["Host IP","Bridge","Client IP","Internet","Tunnel","SSH","IS"]

// ── App Delegate ──────────────────────────────────────────────────
class AppDelegate: NSObject, NSApplicationDelegate {
    var panel: NSPanel!
    var dotFields  = [String: NSTextField]()
    var valFields  = [String: NSTextField]()
    var outputView: NSTextView!
    var scrollView: NSScrollView!

    func applicationDidFinishLaunching(_ n: Notification) {
        buildPanel()
        panel.makeKeyAndOrderFront(nil)
        NSApp.activate(ignoringOtherApps: true)
        Timer.scheduledTimer(withTimeInterval: REFRESH_S, repeats: true) { _ in
            if let s = getStatus() { self.refreshStatus(s) }
        }
    }

    func buildPanel() {
        let style: NSWindow.StyleMask = [.titled,.closable,.miniaturizable,
                                         .resizable,.utilityWindow]
        panel = NSPanel(contentRect: NSRect(x:200,y:200,width:W,height:H),
                        styleMask: style,
                        backing: .buffered, defer: false)
        panel.title = "Bridge Restore"
        panel.backgroundColor = cBg
        panel.level = .floating
        panel.isFloatingPanel = true
        panel.becomesKeyOnlyIfNeeded = true
        panel.hidesOnDeactivate = false
        panel.isReleasedWhenClosed = false

        let cv = panel.contentView!
        buildLeft(cv)
        buildDivider(cv)
        buildOutputPane(cv)
        populateWelcome()
    }

    func buildLeft(_ parent: NSView) {
        let bg = NSView(frame: NSRect(x:0,y:0,width:LW,height:H))
        bg.wantsLayer = true; bg.layer?.backgroundColor = cPanel.cgColor
        parent.addSubview(bg)

        // Header
        let mono12 = NSFont.monospacedSystemFont(ofSize:12, weight:.regular)
        let sys11  = NSFont.systemFont(ofSize:11)
        let sysBold14 = NSFont.boldSystemFont(ofSize:14)
        let sys12  = NSFont.systemFont(ofSize:12)
        let mono11 = NSFont.monospacedSystemFont(ofSize:11, weight:.regular)

        let hdr = makeLabel("BRIDGE RESTORE", sysBold14, cWhite)
        hdr.frame = NSRect(x:PAD, y:H-36, width:LW-PAD*2, height:22)
        bg.addSubview(hdr)

        let st = getStatus()
        let roleStr = st.map { "\($0.role): \($0.friendly)  ·  \($0.model)" } ?? "checking…"
        let sub = makeLabel(roleStr, sys11, cGrey)
        sub.frame = NSRect(x:PAD, y:H-56, width:LW-PAD*2, height:16)
        bg.addSubview(sub)

        // Separator
        let sep = NSBox(frame: NSRect(x:PAD,y:H-68,width:LW-PAD*2,height:1))
        sep.boxType = .separator; bg.addSubview(sep)

        // Status rows
        var y = H - 92
        for (i, key) in ROW_KEYS.enumerated() {
            let chk = st?.checks[key]
            let dot = makeLabel(dotChar(chk?.status ?? "grey"), mono12,
                                dotColor(chk?.status ?? "grey"))
            dot.frame = NSRect(x:PAD, y:y, width:18, height:18)
            bg.addSubview(dot); dotFields[key] = dot

            let lbl = makeLabel(ROW_LABELS[i], sys11, cDim)
            lbl.frame = NSRect(x:PAD+22, y:y, width:90, height:16)
            bg.addSubview(lbl)

            let val = makeLabel(chk?.value ?? "…", mono11, cWhite)
            val.frame = NSRect(x:PAD+116, y:y, width:LW-PAD-122, height:16)
            bg.addSubview(val); valFields[key] = val
            y -= 24
        }

        // Separator above buttons
        let sep2 = NSBox(frame: NSRect(x:PAD,y:168,width:LW-PAD*2,height:1))
        sep2.boxType = .separator; bg.addSubview(sep2)

        // Buttons
        let BW = (LW - PAD*3) / 2
        let btns: [(String, String)] = [
            ("tunnel-mini",  "tunnel-mini"),
            ("tunnel-pro",   "tunnel-pro"),
            ("Status",       STATUS_CMD),
            ("Diagnostics",  "\(SCRIPTS)/bridge-restore-app.sh --diag"),
            ("Console",      "log show --last 10m --predicate 'messageType == fault OR messageType == error' --style compact 2>/dev/null | grep -v '^Timestamp' | tail -40 || echo 'No errors in last 10m'"),
            ("Health",       "ps -axro %cpu,%mem,pid,user,comm | head -12; echo '---'; vm_stat | grep -E 'free|active|inactive|wired'; echo '---'; uptime"),
            ("Rescue Bot",   "rescue-bot diagnose"),
            ("Refresh",      STATUS_CMD)
        ]
        btns.enumerated().forEach { (i, btn) in
            let col = i % 2; let row = i / 2
            let bx = PAD + CGFloat(col)*(BW+PAD)
            let by = CGFloat(148 - row*36)
            let b = NSButton(frame: NSRect(x:bx,y:by,width:BW,height:28))
            b.isBordered = false
            b.wantsLayer = true
            b.layer?.backgroundColor = NSColor(red:58/255,green:58/255,blue:60/255,alpha:1).cgColor
            b.layer?.cornerRadius = 6
            b.layer?.borderWidth = 0.5
            b.layer?.borderColor = NSColor(red:100/255,green:100/255,blue:105/255,alpha:1).cgColor
            b.attributedTitle = NSAttributedString(string: btn.0, attributes: [
                .font: sys12,
                .foregroundColor: NSColor.white
            ])
            b.target = self
            b.action = #selector(buttonAction(_:))
            b.identifier = NSUserInterfaceItemIdentifier(btn.1)
            bg.addSubview(b)
        }

        // IS Settings shortcut
        let isBtn = NSButton(frame: NSRect(x:PAD,y:10,width:LW-PAD*2,height:24))
        isBtn.isBordered = false
        isBtn.wantsLayer = true
        isBtn.layer?.backgroundColor = NSColor(red:80/255,green:50/255,blue:10/255,alpha:1).cgColor
        isBtn.layer?.cornerRadius = 6
        isBtn.layer?.borderWidth = 0.5
        isBtn.layer?.borderColor = cYellow.cgColor
        isBtn.attributedTitle = NSAttributedString(string: "⚠  Open Sharing Settings", attributes: [
            .font: sys11,
            .foregroundColor: cYellow
        ])
        isBtn.target = self; isBtn.action = #selector(openSharing)
        bg.addSubview(isBtn)
    }

    func buildDivider(_ cv: NSView) {
        let d = NSBox(frame: NSRect(x:LW,y:0,width:1,height:H))
        d.boxType = .separator; cv.addSubview(d)
    }

    func buildOutputPane(_ cv: NSView) {
        scrollView = NSScrollView(frame: NSRect(x:LW+1,y:0,width:W-LW-1,height:H))
        scrollView.hasVerticalScroller = true
        scrollView.autohidesScrollers = true
        scrollView.backgroundColor = cOutBg

        outputView = NSTextView(frame: NSRect(x:0,y:0,width:W-LW-1,height:H))
        outputView.backgroundColor = cOutBg
        outputView.textColor = cWhite
        outputView.font = NSFont.monospacedSystemFont(ofSize:11, weight:.regular)
        outputView.isEditable = false
        outputView.isRichText = true
        outputView.isAutomaticQuoteSubstitutionEnabled = false

        scrollView.documentView = outputView
        cv.addSubview(scrollView)
    }

    func appendOutput(_ text: String, _ color: NSColor = .white) {
        let attrs: [NSAttributedString.Key: Any] = [
            .font: NSFont.monospacedSystemFont(ofSize:11, weight:.regular),
            .foregroundColor: color
        ]
        let astr = NSAttributedString(string: text + "\n", attributes: attrs)
        outputView.textStorage?.append(astr)
        outputView.scrollRangeToVisible(NSRange(location: outputView.string.count, length: 0))
    }

    func populateWelcome() {
        appendOutput("Bridge Restore v1.0.6 (Build 26A06)", cBlue)
        appendOutput("─────────────────────────────────────────", cBorder)
        if let s = getStatus() {
            let msg: String
            let col: NSColor
            if s.summary.red_count > 0 {
                msg = "⚠️  \(s.summary.red_count) check(s) need attention"
                col = cYellow
            } else if s.summary.grey_count > 0 {
                let active = 7 - s.summary.grey_count
                msg = "✅ \(active)/7 active checks green  ·  \(s.summary.grey_count) N/A for this role"
                col = cGreen
            } else {
                msg = "✅ All systems green"
                col = cGreen
            }
            appendOutput(msg, col)
        }
        appendOutput("Press a button to run a command.", cDim)
        appendOutput("Output streams here. Refreshes every \(Int(REFRESH_S))s.", cDim)
    }

    func refreshStatus(_ s: Status) {
        DispatchQueue.main.async {
            for key in ROW_KEYS {
                guard let chk = s.checks[key] else { continue }
                self.dotFields[key]?.textColor = dotColor(chk.status)
                self.dotFields[key]?.stringValue = dotChar(chk.status)
                self.valFields[key]?.stringValue = chk.value
            }
        }
    }

    @objc func buttonAction(_ sender: NSButton) {
        let cmd = sender.identifier?.rawValue ?? ""
        appendOutput("\n▶ \(sender.title)", cBlue)
        appendOutput("  $ \(cmd)", cDim)
        DispatchQueue.global(qos: .userInitiated).async {
            let out = shell(cmd)
            DispatchQueue.main.async {
                out.components(separatedBy: "\n").forEach { line in
                    let col: NSColor = line.contains("✅") || line.contains("OK") ? cGreen :
                                       line.contains("⚠️") || line.contains("WARN") ? cYellow :
                                       line.contains("❌") || line.contains("FAIL") ? cRed : cWhite
                    self.appendOutput(line, col)
                }
                self.appendOutput("─────────────────────────────────────", cBorder)
                if let s = getStatus() { self.refreshStatus(s) }
            }
        }
    }

    @objc func openSharing() {
        NSWorkspace.shared.open(
            URL(string:"x-apple.systempreferences:com.apple.preferences.sharing?Internet_Sharing")!)
        appendOutput("Opened System Settings → Sharing", cYellow)
    }
}

// ── Entry point ───────────────────────────────────────────────────
let app = NSApplication.shared
app.setActivationPolicy(.accessory)
let delegate = AppDelegate()
app.delegate = delegate
app.run()

File: app_macOS-Intel_BridgeRestore/scripts/bridge-restore-panel.js
#!/usr/bin/env osascript -l JavaScript
// ╔══════════════════════════════════════════════════════════════════╗
// ║  bridge-restore-panel.js — Phase 1b persistent NSPanel          ║
// ║  JXA + ObjC bridge: floating dashboard, never disappears        ║
// ║  VERSION: v1.0.6 (Build 26A06)                            ║
// ╚══════════════════════════════════════════════════════════════════╝

ObjC.import('AppKit')
ObjC.import('Foundation')

const SCRIPTS = '/Users/akmacks/dev/projects/apps/app_macOS-Intel_BridgeRestore/scripts'
const STATUS_CMD  = `bash ${SCRIPTS}/status-check.sh`
const REFRESH_S   = 15
const W = 820; const H = 520
const LEFT_W = 360; const PAD = 12

// ── Colours ────────────────────────────────────────────────────────
function rgb(r,g,b) {
  return $.NSColor.colorWithRedGreenBlueAlpha(r/255, g/255, b/255, 1)
}
const C = {
  bg:      rgb(28,  28,  30),
  panel:   rgb(44,  44,  46),
  border:  rgb(72,  72,  74),
  green:   rgb(48, 209,  88),
  yellow:  rgb(255,214,  10),
  red:     rgb(255,  69,  58),
  grey:    rgb(142, 142, 147),
  blue:    rgb(10,  132, 255),
  white:   rgb(255, 255, 255),
  dim:     rgb(180, 180, 185),
  btnBg:   rgb(58,  58,  60),
  outBg:   rgb(20,  20,  22),
}

// ── Helper: make NSFont ────────────────────────────────────────────
const mono  = (sz) => $.NSFont.fontWithNameSize('Menlo', sz)
const sans  = (sz) => $.NSFont.fontWithNameSize('SF Pro Display', sz) ||
                      $.NSFont.systemFontOfSize(sz)
const sansBold = (sz) => $.NSFont.boldSystemFontOfSize(sz)

// ── Helper: NSTextField label ──────────────────────────────────────
function label(text, x, y, w, h, font, color) {
  const f = $.NSTextField.alloc.initWithFrame($.NSMakeRect(x, y, w, h))
  f.stringValue = text
  f.font = font
  f.textColor = color
  f.backgroundColor = $.NSColor.clearColor
  f.bezeled = false
  f.editable = false
  f.selectable = false
  return f
}

// ── Helper: run shell command, return stdout ───────────────────────
function shell(cmd) {
  const t = $.NSTask.alloc.init
  t.launchPath = '/bin/zsh'
  t.arguments  = ['-c', cmd]
  const p = $.NSPipe.pipe
  t.standardOutput = p
  t.standardError  = $.NSPipe.pipe
  t.launch; t.waitUntilExit
  const d = p.fileHandleForReading.readDataToEndOfFile
  return $.NSString.alloc.initWithDataEncoding(d, $.NSUTF8StringEncoding).js
}

// ── Parse status JSON ──────────────────────────────────────────────
function getStatus() {
  try {
    const raw = shell(STATUS_CMD)
    return JSON.parse(raw)
  } catch(e) {
    return null
  }
}

// ── Colour for status string ───────────────────────────────────────
function statusColor(s) {
  if (s === 'green')  return C.green
  if (s === 'yellow') return C.yellow
  if (s === 'red')    return C.red
  return C.grey
}

// ── Dot character for status ───────────────────────────────────────
function dot(s) {
  if (s === 'green')  return '●'
  if (s === 'yellow') return '◐'
  if (s === 'red')    return '●'
  return '○'
}

// ── Build status rows in the left panel ───────────────────────────
let dotFields = {}   // key → NSTextField (dot)
let valFields = {}   // key → NSTextField (value)

function buildStatusRows(parent, status) {
  const rows = [
    { key: 'host_ip',       label: 'Host IP' },
    { key: 'bridge',        label: 'Bridge' },
    { key: 'peer_ip',       label: 'Client IP' },
    { key: 'internet',      label: 'Internet' },
    { key: 'tunnel',        label: 'Tunnel' },
    { key: 'controlmaster', label: 'SSH' },
    { key: 'is',            label: 'IS' },
  ]
  let y = H - 160
  rows.forEach(r => {
    const chk = status ? status.checks[r.key] : null
    const st  = chk ? chk.status : 'grey'
    const val = chk ? chk.value  : '…'

    // dot
    const df = label(dot(st), PAD, y, 20, 20, mono(14), statusColor(st))
    parent.addSubview(df); dotFields[r.key] = df

    // row label
    parent.addSubview(label(r.label, PAD+22, y, 90, 18, sans(12), C.dim))

    // value
    const vf = label(val, PAD+114, y, LEFT_W-PAD-120, 18, mono(12), C.white)
    parent.addSubview(vf); valFields[r.key] = vf

    y -= 24
  })
}

// ── Update status rows (called on refresh) ────────────────────────
function refreshStatus(status) {
  if (!status) return
  Object.keys(dotFields).forEach(k => {
    const chk = status.checks[k]
    if (!chk) return
    dotFields[k].textColor = statusColor(chk.status)
    dotFields[k].stringValue = dot(chk.status)
    valFields[k].stringValue = chk.value
  })
}

// ── NSButton helper ────────────────────────────────────────────────
function makeButton(title, x, y, w, h, action) {
  const b = $.NSButton.alloc.initWithFrame($.NSMakeRect(x, y, w, h))
  b.title = title
  b.bezelStyle = $.NSBezelStyleRounded
  b.font = sans(12)
  b.target = action
  b.action = 'callAsFunction'
  return b
}

// ── Output NSTextView (right panel) ───────────────────────────────
let outputTV

function buildOutputPane(parent) {
  const scroll = $.NSScrollView.alloc.initWithFrame(
    $.NSMakeRect(LEFT_W+1, 0, W-LEFT_W-1, H))
  scroll.hasVerticalScroller = true
  scroll.autohidesScrollers = true
  scroll.backgroundColor = C.outBg

  outputTV = $.NSTextView.alloc.initWithFrame(
    $.NSMakeRect(0, 0, W-LEFT_W-1, H))
  outputTV.backgroundColor = C.outBg
  outputTV.textColor = C.white
  outputTV.font = mono(11)
  outputTV.editable = false
  outputTV.richText = false
  outputTV.automaticQuoteSubstitutionEnabled = false

  scroll.documentView = outputTV
  parent.addSubview(scroll)
}

// ── Append coloured text to output pane ───────────────────────────
function appendOutput(text, color) {
  const col = color || C.white
  const attrs = $.NSMutableDictionary.alloc.init
  attrs.setObjectForKey(mono(11), $.NSFontAttributeName)
  attrs.setObjectForKey(col, $.NSForegroundColorAttributeName)
  const astr = $.NSAttributedString.alloc.initWithStringAttributes(text+'\n', attrs)
  outputTV.textStorage.appendAttributedString(astr)
  outputTV.scrollRangeToVisible(
    $.NSMakeRange(outputTV.string.length, 0))
}

// ── Run command, stream to output pane ────────────────────────────
function runCmd(label, cmd) {
  appendOutput(`\n▶ ${label}`, C.blue)
  appendOutput(`  $ ${cmd}`, C.dim)
  const out = shell(cmd)
  out.split('\n').forEach(line => {
    const col = line.includes('✅') || line.includes('OK') ? C.green :
                line.includes('⚠️') || line.includes('WARN') ? C.yellow :
                line.includes('❌') || line.includes('FAIL') ? C.red : C.white
    appendOutput(line, col)
  })
  appendOutput('─────────────────────────────────────', C.border)
}

// ══════════════════════════════════════════════════════════════════
// MAIN — build and show the panel
// ══════════════════════════════════════════════════════════════════

const app = $.NSApplication.sharedApplication
app.setActivationPolicy($.NSApplicationActivationPolicyAccessory)

// ── Create the NSPanel ─────────────────────────────────────────────
const styleMask = $.NSWindowStyleMaskTitled
  | $.NSWindowStyleMaskClosable
  | $.NSWindowStyleMaskMiniaturizable
  | $.NSWindowStyleMaskResizable
  | $.NSWindowStyleMaskUtilityWindow

const panel = $.NSPanel.alloc.initWithContentRectStyleMaskBackingDefer(
  $.NSMakeRect(200, 200, W, H), styleMask,
  $.NSBackingStoreBuffered, false)
panel.title = 'Bridge Restore'
panel.backgroundColor = C.bg
panel.level = $.NSFloatingWindowLevel
panel.floatingPanel = true
panel.becomesKeyOnlyIfNeeded = true
panel.hidesOnDeactivate = false

const content = panel.contentView

// ── Left panel background ──────────────────────────────────────────
const leftBg = $.NSView.alloc.initWithFrame($.NSMakeRect(0, 0, LEFT_W, H))
leftBg.wantsLayer = true
leftBg.layer.backgroundColor = C.panel.CGColor
content.addSubview(leftBg)

// ── Vertical divider ──────────────────────────────────────────────
const div = $.NSBox.alloc.initWithFrame($.NSMakeRect(LEFT_W, 0, 1, H))
div.boxType = $.NSBoxSeparator
content.addSubview(div)

// ── Header: role + friendly name ──────────────────────────────────
const status0 = getStatus()
const role     = status0 ? status0.role     : '…'
const friendly = status0 ? status0.friendly : '…'
const model    = status0 ? status0.model    : '…'

leftBg.addSubview(label('BRIDGE RESTORE', PAD, H-36, LEFT_W-PAD*2, 22,
  sansBold(15), C.white))
leftBg.addSubview(label(`${role}: ${friendly}  ·  ${model}`, PAD, H-56,
  LEFT_W-PAD*2, 16, sans(11), C.grey))

// ── Separator under header ────────────────────────────────────────
const sep1 = $.NSBox.alloc.initWithFrame($.NSMakeRect(PAD, H-70, LEFT_W-PAD*2, 1))
sep1.boxType = $.NSBoxSeparator; leftBg.addSubview(sep1)

// ── Status rows ────────────────────────────────────────────────────
buildStatusRows(leftBg, status0)

// ── Separator above buttons ────────────────────────────────────────
const sep2 = $.NSBox.alloc.initWithFrame($.NSMakeRect(PAD, 170, LEFT_W-PAD*2, 1))
sep2.boxType = $.NSBoxSeparator; leftBg.addSubview(sep2)

// ── Buttons ────────────────────────────────────────────────────────
const BW = (LEFT_W - PAD*3) / 2
const buttons = [
  { label: 'tunnel-mini',  cmd: 'tunnel-mini' },
  { label: 'tunnel-pro',   cmd: 'tunnel-pro'  },
  { label: 'Status',       cmd: `bash ${SCRIPTS}/status-check.sh` },
  { label: 'Diagnostics',  cmd: `bash ${SCRIPTS}/bridge-restore-app.sh --diag` },
  { label: 'Rescue Bot',   cmd: 'rescue-bot diagnose' },
  { label: 'Refresh',      cmd: STATUS_CMD },
]

const btnRows = [[buttons[0],buttons[1]],[buttons[2],buttons[3]],[buttons[4],buttons[5]]]
btnRows.forEach((row, ri) => {
  row.forEach((btn, ci) => {
    const bx = PAD + ci*(BW+PAD)
    const by = 130 - ri*40
    const b = makeButton(btn.label, bx, by, BW, 30, () => {
      appendOutput('', C.dim)
      runCmd(btn.label, btn.cmd)
      if (btn.label === 'Refresh' || btn.label === 'Status') {
        const s = getStatus()
        if (s) refreshStatus(s)
      }
    })
    leftBg.addSubview(b)
  })
})

// ── IS restart helper button ───────────────────────────────────────
const isBtn = makeButton('⚠ Open Sharing Settings', PAD, 18, LEFT_W-PAD*2, 22, () => {
  shell(`open 'x-apple.systempreferences:com.apple.preferences.sharing?Internet_Sharing'`)
  appendOutput('Opened System Settings → Sharing', C.yellow)
})
isBtn.font = sans(11)
leftBg.addSubview(isBtn)

// ── Right output pane ──────────────────────────────────────────────
buildOutputPane(content)

// ── Welcome message in output ─────────────────────────────────────
appendOutput('Bridge Restore v1.0.6 (Build 26A06)', C.blue)
appendOutput('─────────────────────────────────────────', C.border)
if (status0) {
  const s = status0.summary
  const msg = s.red_count === 0
    ? '✅ All systems green'
    : `⚠️  ${s.red_count} check(s) need attention`
  appendOutput(msg, s.red_count === 0 ? C.green : C.yellow)
}
appendOutput('Press a button to run a command. Output appears here.', C.dim)
appendOutput('Status refreshes every 15 seconds.', C.dim)

// ── NSTimer for status refresh every 15s ─────────────────────────
ObjC.registerSubclass({
  name: 'BRRefresher',
  superclass: 'NSObject',
  methods: {
    'refresh:': {
      types: ['void', ['id']],
      implementation: function() {
        const s = getStatus()
        if (s) {
          refreshStatus(s)
          if (s.checks.is && s.checks.is.status === 'red') {
            dotFields['is'].textColor = C.red
            dotFields['is'].stringValue = '●'
          }
        }
      }
    }
  }
})

const refresher = $.BRRefresher.alloc.init
$.NSTimer.scheduledTimerWithTimeIntervalTargetSelectorUserInfoRepeats(
  REFRESH_S, refresher, 'refresh:', $(), true)

// ── Show panel and run event loop ─────────────────────────────────
panel.makeKeyAndOrderFront(null)
app.activateIgnoringOtherApps(true)

// app.run exits immediately in shell context — use NSRunLoop directly
const runLoop = $.NSRunLoop.mainRunLoop
while (true) {
  runLoop.runUntilDate($.NSDate.dateWithTimeIntervalSinceNow(0.5))
}
